CHANGELOG.md

```diff
@@ -2,6 +2,58 @@
 
 This document includes a curated changelog for each release. We also publish a changelog as the description of a [GitHub release](https://github.com/nginxinc/nginx-kubernetes-gateway/releases), which, by contrast, is auto-generated and includes links to all PRs that went into the release.
 
+## Release 0.4.0
+
+*July 5, 2023*
+
+This release brings:
+- Extended support of the features of the Gateway API resources.
+- Support for running the conformance test suite. See the [Conformance tests README](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/v0.4.0/conformance/README.md).
+- Defined Enhancement Proposal process for NGINX Kubernetes Gateway project. See the [Proposal README](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/v0.4.0/docs/proposals/README.md).
```

Comment:
I suggest two more items here, because of their importance:
- Developer documentation for contributing to the project. See the [quick start](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/v0.4.0/docs/developer/quickstart.md).
- Architecture [document](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/main/docs/architecture.md) that explains how NGINX Kubernetes Gateway works at a high level.